plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -115,55 +117,82 @@ public void testNoColumnStats()
     }
 
     @Test
-    public void testCorruptedTableLocation()
+    public void testCorruptedManagedTableLocation()
+            throws Exception
+    {
+        String tableName = "bad_person_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 id, 'person1' name", 1);
+        String tableLocation = (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName);
+        // Remove leading 'file:' because $path column returns 'file:/path-to-file' in case of local file system
+        testCorruptedTableLocation(tableName, Path.of(tableLocation.replaceFirst("^file:", "")), true);
+    }
+
+    @Test
+    public void testCorruptedExternalTableLocation()
             throws Exception
     {
         // create a bad_person table which is based on person table in temporary location
-        String tableName = "bad_person";
-        Path tableLocation = Files.createTempFile("bad_person", null);
+        String tableName = "bad_person_" + randomNameSuffix();
+        Path tableLocation = Files.createTempFile(tableName, null);
         copyDirectoryContents(Path.of(getTableLocation("person").toURI()), tableLocation);
         getQueryRunner().execute(
                 format("CALL system.register_table('%s', '%s', '%s')", getSession().getSchema().orElseThrow(), tableName, tableLocation));
+        testCorruptedTableLocation(tableName, tableLocation, false);
+    }
 
-        // break the table by deleting all its files including transaction log
-        deleteRecursively(tableLocation, ALLOW_INSECURE);
+    private void testCorruptedTableLocation(String tableName, Path tableLocation, boolean isManaged)
+            throws Exception
+    {
+        Path transactionLogDirectory = tableLocation.resolve("_delta_log");
+
+        // break the table by deleting all its files under transaction log
+        deleteRecursively(transactionLogDirectory, ALLOW_INSECURE);
```

Comment:
Previously, the test deleted the whole table location. Could you leave the motivation to the commit body?